includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1248,6 +1261,16 @@ public function get_payment_method() {
 		return $this->payment_methods['card'];
 	}
 
+	/**
+	 * Return the payment method type from the payment method details.
+	 *
+	 * @param array $payment_method_details Payment method details.
+	 * @return string|null Payment method type or nothing.
+	 */
+	public function get_payment_method_type_from_payment_details( $payment_method_details ) {
```

Comment:
I would vote for the most restrictive access level modifier if a method is invoked from inside of the class only.

```suggestion
	private function get_payment_method_type_from_payment_details( $payment_method_details ) {
```

Reply:
Yes, nice catch! Thank again Timur 🚀